File: python/sources/treina2.py
import random
import json

# Lê amostras
with open('entradas2.txt', 'r', encoding='utf-8') as f:
    amostras = json.load(f) # carrega na variavel amostra

# Inicializa os bias com valores aleatorios
bias = [random.random(), random.random(), random.random()]
finalBias = 40 + random.random()

for x in range(1000000): # ajusta os bias milhares de :ezes
    a = amostras[random.randint(0, 14)]  # uma laranja qualquer
    caracteristica = [a['peso'], a['volume'], a['cor'], a['posicao'], a['textura'], a['consistencia'], a['temperatura'], a['idade']]
    referencia = sum(caracteristica)

    for t in range(3): # ajusta os bias da camada intermediaria
        if referencia > bias[t]: # um pouco a mais
            bias[t] += bias[t] * 0.000007
        elif referencia < bias[t]: # um pouco a menos
            bias[t] -= bias[t] * 0.000007

    referenciaFinal = referencia * bias[0] + referencia * bias[1] + referencia * bias[2] # ajusta o bias final

    if referenciaFinal > finalBias: # um pouco a mais
        finalBias += finalBias * 0.000007
    elif referenciaFinal < finalBias: # um pouco a menos
        finalBias -= finalBias * 0.000007

    print(f"finalBias = {finalBias}")

# Salva os bias no arquivo
with open('bias1.txt', 'w', encoding='utf-8') as f:
    f.write('\n'.join([str(b) for b in bias]) + '\n' + str(finalBias) + '\n')

File: nodeJs/sources/treina2.js
const fs = require('fs');

// Lê amostras
const amostras = JSON.parse(fs.readFileSync('entradas2.txt', 'utf-8'));

let bias = [Math.random(), Math.random(), Math.random()];
let finalBias = 40 + Math.random();

for (let x = 0; x < 1000000; x++) { //treina a rede milhares de vezes
  const a = amostras[Math.floor(Math.random() * 15)]; //uma laranja qualquer

  const caracteristica = [a.peso, a.volume, a.cor, a.posicao, a.textura, a.consistencia, a.temperatura, a.idade];

  let referencia = caracteristica.reduce((soma, v) => soma + v, 0);

  for (let t = 0; t < 3; t++) { //ajusta a camada intermediaria
    if (referencia > bias[t]) bias[t] += bias[t] * 0.000007; //um pouco a mais
    else if (referencia < bias[t]) bias[t] -= bias[t] * 0.000007; //um pouco a menos
  }

  let referenciaFinal = referencia * bias[0] + referencia * bias[1] + referencia * bias[2];

  if (referenciaFinal > finalBias) finalBias += finalBias * 0.000007; //ajusta um pouco para cima
  else if (referenciaFinal < finalBias) finalBias -= finalBias * 0.000007; //ajusta um pouco para menos

  console.log(`finalBias = ${finalBias}`);
}

// Salva os bias
fs.writeFileSync('bias1.txt', bias.map(b => b + '').join('\n') + '\n' + finalBias + '\n', 'utf-8');
